Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -0,0 +1,372 @@
+
+namespace UnifTight
+
+theorem eventually_cofinite_indicator (hf : UnifTight f p μ) {ε : ℝ≥0∞} (hε : ε ≠ 0) :
+    ∀ᶠ s in μ.cofinite.smallSets, ∀ i, snorm (s.indicator (f i)) p μ ≤ ε := by
+  rcases hf (pos_iff_ne_zero.2 hε) with ⟨s, hμs, hfs⟩
+  refine (eventually_smallSets' ?_).2 ⟨sᶜ, ?_, fun i ↦ hfs i⟩
+  · intro s t hst ht i
+    exact (snorm_mono <| norm_indicator_le_of_subset hst _).trans (ht i)
+  · rwa [Measure.compl_mem_cofinite, lt_top_iff_ne_top]
+
+protected theorem exists_measurableSet_indicator (hf : UnifTight f p μ) {ε : ℝ≥0∞} (hε : ε ≠ 0) :
+    ∃ s, MeasurableSet s ∧ μ s < ∞ ∧ ∀ i, snorm (sᶜ.indicator (f i)) p μ ≤ ε :=
+  let ⟨s, hμs, hsm, hfs⟩ := (hf.eventually_cofinite_indicator hε).exists_measurable_mem_of_smallSets
+  ⟨sᶜ, hsm.compl, hμs, by rwa [compl_compl s]⟩
+
+protected theorem add (hf : UnifTight f p μ) (hg : UnifTight g p μ)
+    (hf_meas : ∀ i, AEStronglyMeasurable (f i) μ) (hg_meas : ∀ i, AEStronglyMeasurable (g i) μ) :
+    UnifTight (f + g) p μ := fun ε hε ↦ by
+  rcases exists_Lp_half β μ p hε.ne' with ⟨η, hη_pos, hη⟩
+  obtain ⟨s, hμs, hsm, hfs, hgs⟩ :
+      ∃ s ∈ μ.cofinite, MeasurableSet s ∧
+        (∀ i, snorm (s.indicator (f i)) p μ ≤ η) ∧ (∀ i, snorm (s.indicator (g i)) p μ ≤ η) :=
+    ((hf.eventually_cofinite_indicator hη_pos.ne').and
+      (hg.eventually_cofinite_indicator hη_pos.ne')).exists_measurable_mem_of_smallSets
+  refine ⟨sᶜ, ne_of_lt hμs, fun i ↦ ?_⟩
+  calc
+    snorm (indicator sᶜᶜ (f i + g i)) p μ = snorm (indicator s (f i) + indicator s (g i)) p μ := by
+      rw [compl_compl, indicator_add']
+    _ ≤ ε := le_of_lt <|
+      hη _ _ ((hf_meas i).indicator hsm) ((hg_meas i).indicator hsm)
+        (hfs i) (hgs i)
+
+protected theorem neg (hf : UnifTight f p μ) : UnifTight (-f) p μ := by
+  simp_rw [UnifTight, Pi.neg_apply, Set.indicator_neg', snorm_neg]
+  exact hf
+
+protected theorem sub (hf : UnifTight f p μ) (hg : UnifTight g p μ)
+    (hf_meas : ∀ i, AEStronglyMeasurable (f i) μ) (hg_meas : ∀ i, AEStronglyMeasurable (g i) μ) :
+    UnifTight (f - g) p μ := by
+  rw [sub_eq_add_neg]
+  exact hf.add hg.neg hf_meas fun i => (hg_meas i).neg
+
+
+protected theorem ae_eq (hf : UnifTight f p μ) (hfg : ∀ n, f n =ᵐ[μ] g n) :
+    UnifTight g p μ := by
+  intro ε hε
+  obtain ⟨s, hμs, hfε⟩ := hf hε
+  refine ⟨s, hμs, fun n => (le_of_eq <| snorm_congr_ae ?_).trans (hfε n)⟩
+  filter_upwards [hfg n] with x hx
+  simp only [indicator, mem_compl_iff, ite_not, hx]
+
+end UnifTight
+
+/-- A constant function is tight. -/
+theorem unifTight_const {g : α → β} (hp_ne_top : p ≠ ∞) (hg : Memℒp g p μ) :
+    UnifTight (fun _ : ι => g) p μ := by
+  intro ε hε
+  by_cases hε_top : ε = ∞
+  · exact ⟨∅, (by measurability), fun _ => hε_top.symm ▸ le_top⟩
+  obtain ⟨s, _, hμs, hgε⟩ := hg.exists_snorm_indicator_compl_lt hp_ne_top hε.ne'
+  exact ⟨s, ne_of_lt hμs, fun _ => hgε.le⟩
+
+/-- A single function is tight. -/
+theorem unifTight_subsingleton [Subsingleton ι] (hp_top : p ≠ ∞)
```

Comment:
```suggestion
theorem unifTight_of_subsingleton [Subsingleton ι] (hp_top : p ≠ ∞)
```

Reply:
The parallel naming convention could also be fixed in UniformIntegrable.lean.